blake2/src/impl_hacl/blake2s.rs

```diff
@@ -0,0 +1,464 @@
+use std::marker::PhantomData;
+
+use crate::hacl::hash_blake2b::{blake2_params, index, params_and_key};
+use crate::hacl::hash_blake2s::{digest, malloc_raw, reset, reset_with_key, state_t, update0};
+
+const PARAM_LEN: usize = 8;
+const MAX_LEN: usize = 32;
+
+use super::*;
+
+/// A builder for [`Blake2s`]. `KEY_LEN`` must be in the range `0..=x61\xa5\x48\xf2\xde\x1c\x31\x8b\xa9\x1d\x52\x07\x00\x78\x61\x01\x0f\x69\xa4\x3e\xc6\x63\xfe\x48\x7d\x84\x03\x28\x2c\x93\x4e\xa7\x25\xdc\x0b\xb1\x72\x25\x6a\xc9\x96\x25\xad\x64\xcc\xa6\xa2\xc4\xd6\x1c\x65\x0a\x35\xaf\xab\x47\x87\xdc\x67\x8e\x19\x07\x1e\xf9``, `OUT_LEN`` must be in the
+/// range `1..=32`.
+pub struct Blake2sBuilder<'a, T> {
+    key: T,
+    personal: &'a [u8; PARAM_LEN],
+    salt: &'a [u8; PARAM_LEN],
+}
+
+impl<'a> Blake2sBuilder<'a, &'a ()> {
+    /// Creates the builder for an unkeyed hasher.
+    pub fn new_unkeyed() -> Result<Self, Error> {
+        Ok(Self {
+            key: &(),
+            personal: &[0; PARAM_LEN],
+            salt: &[0; PARAM_LEN],
+        })
+    }
+
+    /// Constructs the [`Blake2s`] hasher for unkeyed hashes and dynamic digest length.
+    pub fn build(self, digest_length: u8) -> Result<Blake2s<Blake2sStaticKey<0>>, Error> {
```

Comment:
Do I understand correctly that the `Blake2sStaticKey` is only there to hold the key length? Why can't we use a const generic directly for that? Also, we don't really use it.